### ISSUE TITLE
TuriCreate crippled by dependencies on old MXNet

### ISSUE THREAD

**Issue by rplom** [apple/turicreate#2291]
TuriCreate requires MXNet version 1.1.0 it has has serious issues. It's not compatible with fixed version to MXNet version 1.5.0. Because then it needs a new numpy and tensor. Can TuriCreate be updated to not require such outdated frameworks?

TuriCreate is crippled by MXNet and you should change your requirements or else no one will be able to create big models. I can't use the audio classifier past ~60K items before it dies.

see https://github.com/apple/turicreate/issues/2286

**Comment by hoytak** [comment]
Hey @rplom, we are actively working on this issue.  Expect some resolution to it soon.

**Comment by rplom** [comment]
Amazing!  Thank you!  I'm working around it by using random_split with large numbers until I can grow the training set safely again.

**Comment by rplom** [comment]
Just copying some of the MXNet details here from the other ticket 

Large Tensor Support
Before MXNet 1.5.0, MXNet supported a maximal tensor size of around 4 billion (2³²). This was due to uint32_t being used as the default data type for tensor size, as well as variable indexing. Now you can enable large tensor support by changing the following build flag to 1: USE_INT64_TENSOR_SIZE = 1.(Note this is set to 0 by default) This enabled large scale training for example large graph network training using Deep Graph Library.

see: apache/incubator-mxnet#9207
https://medium.com/apache-mxnet/apache-mxnet-1-5-0-release-is-now-available-4138f5233401

**Comment by TobyRoseman** [comment]
TuriCreate is API compatible with MXNet 1.5.0. TuriCreate even use 1.5.0 but only for Python 3.7 with macOS. We couldn't get it to work on Linux. We got nondeterminstic hangs in several but not all of our deep learning toolkits.

@rplom - are you using macOS or Linux? What toolkit are you using (i.e. what method are you calling `.create(...)` on)?

**Comment by rplom** [comment]
macOS X running your standard install with venv 
I followed the requirements you have on the manual page.

**Comment by rplom** [comment]
When I got up around ~62K sounds (I don't know the exact number) I started hitting MXNet integer overflow issues.  I tracked down the issue in MXNet and built a 1.5.0 version with the USE_INT64_TENSOR_SIZE.  As soon as I ran Turicreate I got  warning about my numpy.  I updated that.  The next run I got an error with numpy wanting a new Tensorflow.  I successfully ran the model creation at that point, but I was concerned that even though "it worked" there were warnings and I wasn't sure if that would impact or degrade the model accuracy.  I decided to revert to the prescribed install versions and log this issue here.  I'm working around my upper bound problem by randomly splitting my training set to keep it under the threshold for overflow.  Ideally tho I can foresee 100K sounds to get the breadth I need for my application.

If you think it's safe to run those newer frameworks with Turicreate, then I will update to those.
